widget/entry.go

```diff
@@ -817,6 +831,13 @@ func (e *Entry) cutToClipboard(clipboard fyne.Clipboard) {
 
 	e.copyToClipboard(clipboard)
 	e.setFieldsAndRefresh(e.eraseSelection)
+	e.propertyLock.Lock()
+	content := e.textProvider().String()
+	if e.OnChanged != nil {
+		e.OnChanged(content)
```

Comment:
I realized that this still is creating a new copy of the string. Looking at `eraseSelection`, it calls `e.updateText()` which sets the `e.Text` field and as we are in a lock we can simply reused that instead of recreating the string once more removing the need to allocate a new string.

```suggestion
	if e.OnChanged != nil {
		e.OnChanged(e.Text)
```